app/controllers/katello/api/v2/environments_controller.rb

```diff
@@ -93,12 +94,24 @@ def create
     param :organization_id, :number, :desc => N_("name of the organization")
     param :new_name, String, :desc => N_("new name to be given to the environment")
     param :description, String, :desc => N_("description of the environment")
+    param :registry_name_pattern, String, :desc => N_("pattern for container image names")
+    param :async, :bool, desc: N_("Do not wait for the update action to finish. Default: true")
     def update
-      fail HttpErrors::BadRequest, _("Can't update the '%s' environment") % "Library" if @environment.library?
+      async = ::Foreman::Cast.to_bool(params.fetch(:async, true))
       update_params = environment_params
+      fail HttpErrors::BadRequest, _("Can't update the '%s' environment") % "Library" if @environment.library? && update_params.empty?
       update_params[:name] = params[:environment][:new_name] if params[:environment][:new_name]
       @environment.update_attributes!(update_params)
-      respond
+      if update_params[:registry_name_pattern]
+        task = send(async ? :async_task : :sync_task, ::Actions::Katello::Environment::PublishRepositories,
+                    @environment, content_type: Katello::Repository::DOCKER_TYPE)
+      end
+
+      if params.include?(:async) && async && task
+        respond_for_async(resource: task)
+      else
+        respond
+      end
```

Comment:
@jlsherrill - If the async param is present, has a value of true, and a task was created, then return the task. This seems like a good compromise. If you agree, I will open an issue to enhance the UI to allow the async task to happen inline (progress bar, etc.).